### PR TITLE
Wrap recover

### PR DESCRIPTION
**Pull request by nejisama**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Sign the CLA
Make sure you have signed the [CLA](https://www.clahub.com/agreements/alipay/sofa-mosn)

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, 
details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases.
And you need to show the ut result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/server/handler.go [hunk 1]
```diff
@@ -203,17 +203,15 @@ func (ch *connHandler) StartListener(lctx context.Context, listenerTag uint64) {
...
 		}
 	}
 }
 
 func (ch *connHandler) StartListeners(lctx context.Context) {
 	for _, l := range ch.listeners {
 		// start goroutine
-		go l.listener.Start(lctx)
-		// set listener addr metrics
-		metrics.AddListenerAddr(l.listener.Addr().String())
```
Comment (taoyuanyuan) [inline pkg/server/handler.go]: 这个怎么删了？

File: pkg/utils/syscall.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		fp, err := os.OpenFile(newFilePath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+		if err != nil {
+			return
+		}
+		syscall.Dup2(int(fp.Fd()), int(stdFile.Fd()))
+	}
+	// call
+	hijack()
+	// rotate
+	ticker := time.NewTicker(hijackRotateInterval)
```
Comment (taoyuanyuan) [inline pkg/utils/syscall.go]: 按天轮转更友好一点

File: pkg/utils/syscall.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	for c := range ticker.C {
+		_ = c
+		if err := os.Rename(newFilePath, newFilePath+".old"); err != nil {
```
Comment (taoyuanyuan) [inline pkg/utils/syscall.go]: rename的时候带上日志 比如 log.2018-5-25,  跟其他日志一样的格式

File: pkg/utils/goroutine.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+var debugIgnoreStdout = false
+
+// GoWithRecover wraps a `go func()` with recover()
+func GoWithRecover(handler func(), recoverHandler func(r interface{})) {
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				// TODO: log
+				if !debugIgnoreStdout {
+					fmt.Fprintf(os.Stderr, "goroutine panic: %v\n%s\n", r, string(debug.Stack()))
```
Comment (taoyuanyuan) [inline pkg/utils/goroutine.go]: 为啥要带个 os.Stderr？
